### ISSUE TITLE
Line breaks in attribute tags are effectively disallowed

### ISSUE THREAD

**Issue by benmelz** [sds/haml-lint#612]
I've been aware of this for some time but finally had a quiet weekend morning to look into it:

```haml
%div{wrapped: 'with',
     spaces: 'missing'}
  SpaceInsideHashAttributes: Hash attribute should start with one space after the opening brace
  SpaceInsideHashAttributes: Hash attribute should end with one space before the closing brace or be on its own line

%div{ wrapped: 'with',
      spaces: 'present' }
  RuboCop: Layout/SpaceInsideParens: Space inside parentheses detected.
```

You'll see that when we omit spaces, `haml-lint` complains due to the `SpaceInsideHashLiteralBraces` rule. If we add the spaces it ask for, rubocop then complains due to the `Layout/SpaceInsideParens` rule.

The rubocop offense is odd, but becomes clear when we inspect ruby that is being generated:

```ruby
haml_lint_tag_1
haml_lint_marker_2
WWWW(wrapped: 'with',
     spaces: 'missing')
haml_lint_marker_5
begin
  haml_lint_plain_7
  haml_lint_plain_8
ensure
  HL.noop
end
haml_lint_tag_12
haml_lint_marker_13
WWWW( wrapped: 'with',    # <--- Spaces are preserved
      spaces: 'present' ) # <---
haml_lint_marker_16
begin
  haml_lint_plain_18
ensure
  HL.noop
end
```

---

I'm prepared to submit a fix myself, but I can think of two slightly different paths that could be taken and wanted to see which would be preferable ahead of time.

1. We could make an effort to remove trailing/leading spaces when generating ruby for multiline attribute hashes (which already appears to happen for single-liners).

    This would keep `SpaceInsideHashAttributes` as a concern for `haml-lint` that is separate from rubocop's `Layout/SpaceInsideHashLiteralBraces`.

2. We could preserve the braces when generating ruby for attribute hashes and then remove the `SpaceInsideHashAttributes` rule from `haml-lint` completely.

    This would delegate enforcement of "leading/trailing space inside attribute hashes" to rubocop's `Layout/SpaceInsideHashLiteralBraces` completely.

I personally am leaning towards option 2 - attribute tags look and feel like hashes to me (and unless you're using the esoteric `foo=bar` syntax or `html_attrs()` helper, for all intents and purposes they are). Therefore it makes sense for my preference for hashes in ruby to extend to attribute tags in haml (and as a bonus for `haml-lint`, one less rule to maintain).

That being said if others feel that these should still be individually configurable, I'm fine with implementing option 1.
